ecl/sss/v1/approval_requests/results.go

```diff
@@ -0,0 +1,97 @@
+package approval_requests
+
+import (
+	"github.com/nttcom/eclcloud"
+	"github.com/nttcom/eclcloud/pagination"
+)
+
+type Action struct {
+	Service string `json:"service"`
+	Region  string `json:"region"`
+	APIPath string `json:"api_path"`
+	Method  string `json:"method"`
+	Body    string `json:"body"`
+}
+
+type Description struct {
+	Lang string `json:"lang"`
+	Text string `json:"text"`
+}
+
+// ApprovalRequest represents an ECL SSS Approval Request.
+type ApprovalRequest struct {
+	RequestID         string        `json:"request_id"`
+	ExternalRequestID string        `json:"external_request_id"`
+	ApproverType      string        `json:"approver_type"`
+	ApproverID        string        `json:"approver_id"`
+	RequestUserID     string        `json:"request_user_id"`
+	Service           string        `json:"service"`
+	Actions           []Action      `json:"action"`
+	Descriptions      []Description `json:"descriptions"`
+	RequestUser       interface{}   `json:"request_user"`
+	Approver          bool          `json:"approver"`
+	ApprovalDeadLine  interface{}   `json:"approval_deadline"`
+	ApprovalExpire    interface{}   `json:"approval_expire"`
+	RegisteredTime    interface{}   `json:"registered_time"`
+	UpdatedTime       interface{}   `json:"updated_time"`
+	Status            string        `json:"status"`
+}
+
+type commonResult struct {
+	eclcloud.Result
+}
+
+func (r commonResult) Extract() (*ApprovalRequest, error) {
+	var ar ApprovalRequest
+	err := r.ExtractInto(&ar)
+	return &ar, err
+}
+
+func (r commonResult) ExtractInto(v interface{}) error {
+	return r.Result.ExtractIntoStructPtr(v, "")
+}
+
+// GetResult is the response from a Get operation. Call its Extract method
+// to interpret it as an approval request.
+type GetResult struct {
+	commonResult
+}
+
+// DeleteResult is the response from a Delete operation. Call its ExtractErr to
+// determine if the request succeeded or failed.
+type DeleteResult struct {
+	eclcloud.ErrResult
+}
```

Comment:
The Approval API does not have Delete, but is it necessary to define DeleteResult?

Reply:
You are right. Will fix it.